pkg/ios/xcarchive.go

```diff
@@ -0,0 +1,128 @@
+	"time"
+)
+
+// GetXcodeArchiveLocation retrieves the custom archive location for Xcode archives from user preferences.
+// It first attempts to read the custom location from the system defaults. If the location is not set,
+// it returns the default Xcode archive location within the user's home directory.
+//
+// Parameters:
+// - None
+//
+// Returns:
+// - A string containing the path to the Xcode archive location.
+// - An error if the location cannot be determined or if an error occurs during command execution.
+func GetXcodeArchiveLocation() (string, error) {
+	// Command to read the custom archive location from Xcode preferences
+	cmd := exec.Command("defaults", "read", "com.apple.dt.Xcode", "IDECustomDistributionArchivesLocation")
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+
+	// Execute the command and capture the output
+	err := cmd.Run()
+	if err != nil {
+		// If the command fails, check stderr for additional details
+		if strings.Contains(stderr.String(), "does not exist") {
+			// If the key is not set, return the default location for Xcode archives
+			usr, err := user.Current()
+			if err != nil {
+				return "", fmt.Errorf("unable to get current user: %w", err)
+			}
+			// Return default archive location under user's home directory
+			return filepath.Join(usr.HomeDir, "Library", "Developer", "Xcode", "Archives"), nil
+		}
+		// Return error with additional details from stderr if the command fails
+		return "", fmt.Errorf("error running defaults command: %w, stderr: %s", err, stderr.String())
+	}
+
+	// Trim any trailing newline or spaces from the output
+	customPath := strings.TrimSpace(out.String())
+	if customPath == "" {
+		// Return error if the command succeeded but no path was returned
+		return "", fmt.Errorf("command succeeded but returned an empty path")
+	}
+
+	return customPath, nil
+}
+
+// GetLatestXcodeArchive finds the latest .xcarchive file for a given scheme in the specified directory.
+// It walks through the directory, looking for .xcarchive files that match the scheme. The most recently modified
+// archive file is returned.
+//
+// Parameters:
+// - path: Directory path to search for .xcarchive files.
+// - scheme: The scheme used as a prefix to filter relevant archive files.
+//
+// Returns:
+// - A string containing the path to the most recent .xcarchive file matching the scheme.
+// - An error if no matching archive is found or if an error occurs during the search.
+func GetLatestXcodeArchive(path, scheme string) (string, error) {
+	// Attempt to find the latest .xcarchive for the given scheme in the folder
+	path, err := findLatestXCArchive(path, scheme)
+	if err != nil {
+		return "", err
+	} else if path == "" {
+		// If no archive is found, return an error indicating the folder doesn't contain any matching archives
+		return "", fmt.Errorf("No xcarchive found in %s", path)
+	}
+	// Return the path of the latest archive found
+	return path, nil
+}
```

Comment:
Is this function providing any purpose, other than wrapping `findLatestXCArchive`? Feels like they could easily be combined.

Reply:
You are correct, I've combined the two